gpt_index/indices/query/query_runner.py

```diff
@@ -163,10 +165,15 @@ def _get_query_obj(
         from gpt_index.indices.registry import INDEX_STRUT_TYPE_TO_QUERY_MAP
 
         query_cls = INDEX_STRUT_TYPE_TO_QUERY_MAP[index_struct_type][mode]
-        query_kwargs = self._get_query_kwargs(config)
+        query_kwargs = self._get_query_kwargs(config, additional_query_context)
+
+        additional_query_context = self._additional_query_context[index_struct.index_id]
+        query_kwargs.update(additional_query_context)
```

Comment:
so help me understand: 

currently when the user wants to pass in a vector store (say a pinecone vector store) during query time, they can pass it in in the "query_kwargs" for "pinecone" in the query_configs (e.g. `graph.query(..., query_configs=query_configs)`). Here are you basically proposing that, instead of doing that, there's a certain chunk of arguments that should be passed when the graph is instantiated? 

e.g 
```
ComposableGraph.from_indices(..., additional_query_context=additional_query_context)
```

Reply:
yes that's correct.

I find it strange that we can override the vector store at query time, after we have already "built" an vector store index.
Conceptually, I think only configs should be overridden from kwargs.